bootstrap/config/overlays/ksonnet/kfctl_default.yaml

```diff
@@ -4,10 +4,37 @@ metadata:
   name: config
 spec:
   componentParams:
+    argo:
+    - initRequired: true
```

Comment:
these componentParams were merged to master under [basic-auth](https://github.com/kubeflow/kubeflow/blob/master/bootstrap/config/overlays/basic_auth/kfctl_default.yaml#L10) and [gcp](https://github.com/kubeflow/kubeflow/blob/master/bootstrap/config/overlays/gcp/kfctl_default.yaml#L7)

Reply:
The `gcp` overlay is, well, GCP-specific.
The `basic-auth` overlay is also GCP-specific at the moment.

In general, I don't think it makes sense for those componentParams to be elsewhere.
They are the absolutely necessary componentParams that should accompany the included components.
They should be in the default ksonnet overlay.

This also shouldn't break `basic-auth` or `gcp`.

Reply:
@yanniszark See my comment in https://github.com/kubeflow/kubeflow/issues/2685 about not using the overlays to generate the KfDef YAML files but instead using a self contained YAML file for each platform/config combo we want to support with a clear set of owners.